src/analysis/sensitivity.jl

```diff
@@ -96,13 +133,23 @@ function pawn(X::T1, y::T2, factor_names::Vector{String}; S::Int64=10)::NamedDim
 
     return NamedDimsArray(results; factors=Symbol.(factor_names), Si=[:min, :mean, :median, :max, :std, :cv])
 end
-function pawn(X::DataFrame, y::AbstractVector; S::Int64=10)::NamedDimsArray
+function pawn(X::DataFrame, y::AbstractVector{<:Real}; S::Int64=10)::NamedDimsArray
     return pawn(Matrix(X), y, names(X); S=S)
 end
-function pawn(X::NamedDimsArray, y::T; S::Int64=10)::NamedDimsArray where {T<:Union{NamedDimsArray,AbstractVector{<:Real}}}
+function pawn(X::NamedDimsArray, y::Union{NamedDimsArray,AbstractVector{<:Real}}; S::Int64=10)::NamedDimsArray
     return pawn(X, y, axiskeys(X, 2); S=S)
 end
-function pawn(rs::ResultSet, y::T; S::Int64=10)::NamedDimsArray where {T<:Union{NamedDimsArray,AbstractVector{<:Real}}}
+function pawn(X::Union{DataFrame,AbstractMatrix{<:Real}}, y::AbstractMatrix{<:Real}; S::Int64=10)::NamedDimsArray
```

Comment:
What happens if `X` and `y` are both `AbstractMatrix`? Then this function will call pawn passing `X::AbstractMatrix` and `y::Vector` without `factor_names`, right? Am I missing something or that won't work?

Reply:
You're right - it should constrained to `NamedDimsArray`, not `AbstractMatrix`.